lib/rules/prefer-object-spread.js

```diff
@@ -196,6 +196,34 @@ function autofixObjectLiteral(node, sourceCode) {
     };
 }
 
+/**
+ * Check if the node has modified a given variable
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @returns {boolean} - true if node is an assignment, variable declaration, or import statement
+ */
+function isModifier(node) {
+    if (!node.type) {
+        return false;
+    }
+
+    return node.type === "AssignmentExpression" ||
+    node.type === "VariableDeclarator" ||
+    node.type === "ImportDeclaration";
+}
+
+/**
+ * Check if the node has modified a given variable
+ * @param {array} references - list of reference nodes
+ * @returns {boolean} - true if node is has been overwritten by an assignment or import
+ */
+function overwritingNativeObject(references) {
```

Comment:
This function isn't checking specifically that native Object is being overwritten, right? It just checks that any of the references passed in are being written?

Reply:
yes it might be better to name it `modifyingObjectReference` or something